### PR TITLE
Add Check for Goroutines Count

### LINKED ISSUE #2606: Add /healthz check for goroutine count
Let's return a service `/healthz` error if the number of goroutines reaches some threshold.

Using [runtime.NumGoroutine()](https://golang.org/pkg/runtime/#NumGoroutine), check that the number of routines is under some threshold configured by flag.

`-max-healthy-goroutines=15000` is an example flag to add in which we set 15k to be the default upper limit.

### PR DESCRIPTION
**Pull request by nisdas**
Resolves #2606

### CODE REVIEW

File: beacon-chain/main.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/prysmaticlabs/prysm/shared/version"
 	"github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
-	"github.com/x-cray/logrus-prefixed-formatter"
+	prefixed "github.com/x-cray/logrus-prefixed-formatter"
```
Comment (shayzluf) [inline beacon-chain/main.go]: why did you change the import name?

File: beacon-chain/blockchain/service.go [hunk 2]
```diff
@@ -195,6 +199,9 @@ func (c *ChainService) Stop() error {
 // Status always returns nil.
 // TODO(1202): Add service health checks.
 func (c *ChainService) Status() error {
```
Comment (shayzluf) [inline beacon-chain/blockchain/service.go]: who call status and what are we doing with its response

Comment (nisdas) [reply beacon-chain/blockchain/service.go]: https://github.com/prysmaticlabs/prysm/blob/master/shared/prometheus/service.go#L47

Comment (shayzluf) [reply beacon-chain/blockchain/service.go]: lg

File: shared/cmd/flags.go [hunk 3]
```diff
@@ -105,4 +105,10 @@ var (
 		Usage: "Specify log formatting. Supports: text, json, fluentd.",
 		Value: "text",
 	}
+	// MaxGoroutines specifies the maximum amount of goroutines tolerated, before a status check fails.
+	MaxGoroutines = cli.Int64Flag{
+		Name:  "max-healthy-goroutines",
```
Comment (terencechain) [inline shared/cmd/flags.go]: why not just `max-goroutines`? not sure if `healthy` is needed there

Comment (rauljordan) [reply shared/cmd/flags.go]: +1

Comment (prestonvanloon) [reply shared/cmd/flags.go]:
```suggestion
		Name:  "max-goroutines",
```

Sorry, I was very tired when I suggested that name lol
